src/keylessBackup/saga.ts

```diff
@@ -110,8 +118,48 @@ function* handleKeylessBackupRestore({
     encryptionAddress,
   })
 
-  // TODO(ACT-778): use mnemonic
-  yield* call(decryptPassphrase, torusKeyshareBuffer, valoraKeyshareBuffer, encryptedMnemonic)
+  const decryptedMnemonic = yield* call(
+    decryptPassphrase,
+    torusKeyshareBuffer,
+    valoraKeyshareBuffer,
+    encryptedMnemonic
+  )
+
+  const { privateKey } = yield* call(generateKeysFromMnemonic, decryptedMnemonic)
+  if (!privateKey) {
+    throw new Error('Failed to convert mnemonic to hex')
+  }
+  const backupAccount = privateKeyToAddress(privateKey)
+  if (!(yield* call(walletHasBalance, backupAccount))) {
+    // show zero balance modal
+    yield* put(keylessBackupShowZeroBalance())
+
+    // wait for user to click continue or bail from keyless backup
+    const [_continueAction, bailAction] = yield* race([
```

Comment:
There's no way for a user to exist from the dialog _without_ dispatching one of these actions, right? From the look of it, it seems like not, but wanted to check.

Reply:
nope, no other way to move forward

Reply:
will this resume properly if the app is backgrounded when the modal is visible? What about background press?

Reply:
Just tested, backgroud press does nothing. App being backgrounded also has no effect.